src/router.js

```diff
@@ -35,6 +36,35 @@ const logoutAndRedirect = async (to, from, next) => {
   return next('/login');
 };
 
+const checkAndSubscribeToLobby = async (to, from, next) => {
+  const gameStore = useGameStore();
+  const authStore = useAuthStore();
+  const gameId = parseInt(to.params.gameId);
+  
+  if (Number.isNaN(gameId) || !Number.isFinite(gameId)) {
+    to.meta.error = 'Invalid lobby number';
+    return next();
+  }
+
+  if (!authStore.authenticated) {
+    return next(`/login/${gameId}`);
+  }
+
+  if (gameStore.players.some(({username}) => username === authStore.username)) {
+    return next();
+  }
+  
+  try {
+    await gameStore.requestSubscribe(+gameId);
```

Comment:
Do we still need to cast this if we already called parseInt() above?

Reply:
nope, good catch